### ISSUE TITLE
Uploader not working

### ISSUE THREAD

**Issue by jmatos** [SqrTT/prophet#258]
**Describe the bug**
Uploader is not working
This is what I see in the output window:

Connection validated successfully
Current active version is: my_code
Start uploading cartridges
Cleanup code version...

**To Reproduce**
1. Try to upload 
2. See error

**Expected behavior**
Expected the files to upload

**Screenshots**
Not necessary

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.6
 - VSCode version: Version: 1.49.0 
 - The extension version v1.3.23

**Additional context**

**Comment by jmatos** [comment]
I updated the "cartridge" entry in my dw.json to be limited to just 8 cartridges and that fixed my issue.  The limit appears to be 9 for me, if I specify 10 or more then the upload never completes - I get stuck on the "Cleanup code version..." step.

I suspect if this is not reproducible that it might be related to system resources - I notice that the upload appears to run asynchronously and in parallel - maybe if the upload process uploaded each cartridge serially and not in parallel then this issue might be avoided?

Jorge

**Comment by SqrTT** [comment]
Hi @jmatos thanks reporting.
Indeed I'm not able to reproduce. Some my colleague have said about similar issue and it was fixed by code version remove and recreate (on a sandbox). also try to reboot.

Could you try if it solve the issue in your case, please?

Also would be great to try to connect to another sandbox to see if it's issue is related to sandbox or your workstations.

Anyway most likely issue is with the extension because it should never stuck. It should report error or continue successfully but never stuck. I'll take a look for potential places.

PS.  serial VS  parallel, many people saying that cartridge upload is to long (and took around 1 minute), if we make it a serial this will increase upload time.  Basically this will make many people unhappy :)

**Comment by jmatos** [comment]
Yes - I agree about the serial vs parallel.

I tried rebooting sandbox, removing/recreating code version, renaming code version - nothing worked.

But the extension worked fine when I used a sandbox with no catalogs on it - the upload was very fast actually.  It's weird because my sandbox only has 6 catalogs, 3 master and 3 navigation (3 web sites).  I have a small number of products (~ 1000) across the master catalogs but I'm beginning to suspect that even that number is too much for these sandboxes to work efficiently.

I guess I'll have to decrease the number of products even further to increase my SB performance.

Thanks @SqrTT
